src/main/java/org/jvnet/hudson/test/TemporaryDirectoryAllocator.java

```diff
@@ -85,25 +93,12 @@ public synchronized File allocate() throws IOException {
      * Deletes all allocated temporary directories.
      */
     public synchronized void dispose() throws IOException, InterruptedException {
-        // TODO when we bump the Jenkins dependency to 2.157
-        IOException x = null;
+        System.gc();
```

Comment:
requireing gc that is indicitive of a bug in the code. (something has not been closed yet has gone out of scope so close is called in the finalize.  would suggest not putting this here.

Reply:
Well `DisablePluginCommandTest` reliably fails without this.

Reply:
I'm against adding a call to gc here. The whole point of #166 was to detect when files are not properly closed. Explicit call to gc would hide bugs.

Reply:
…and I am not sure how to track down what got released in finalization. [`file-leak-detector`](https://file-leak-detector.kohsuke.org/) is close but not exactly what we want.

Reply:
jenkinsci/git-plugin#798 <-- these bugs would not be detected if there was GC call.

Reply:
Fine; any suggestions on how to diagnose though? The errors are of various plugin JARs having a mandatory file lock. I know that is generally due to a `JarFile` or a `URLClassLoader` not being `close`d, but there are a lot of JAR accesses in the Jenkins codebase.

Reply:
I would check where the files are opened first - I am guessing that one of them may not be expected JarURLCLassLoader?

https://github.com/jenkinsci/jenkins-test-harness/blob/93e9b64a89d3048896f77d62fba84fff715b932b/src/main/java/org/jvnet/hudson/test/JenkinsRule.java#L363-L373

I can not reproduce locally issues with `DisablePluginCommandTest` (so either yay for timing or a bug in a specific JDK)